pydantic/datetime_parse.py

```diff
@@ -228,6 +230,8 @@ def parse_duration(value: StrBytesIntFloat) -> timedelta:
         value = value.decode()
 
     try:
+        if scientific_notation_duration_re.match(value):
```

Comment:
This looks like it's being run on every attempt to parse a timedelta. Seems like a chunk regex to run on everything just for the corner case. And we pass the value to the float constructor.

Wouldn't it be easier to do `try .. float(...) Except ValueError: pass` to skip the regex algoether?

Reply:
The regex allows to make a first control and to avoid exceptions every time

Reply:
With the regex:
```
In [1]: from datetime import timedelta
In [2]: from pydantic import BaseModel
In [3]: class Foo(BaseModel):    x: timedelta
In [4]: %timeit Foo.parse_raw('{"x": "3e4"}')
15.4 µs ± 165 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

Using just `float(value)` with a try-except block:

```
In [1]: from datetime import timedelta
In [2]: from pydantic import BaseModel
In [3]: class Foo(BaseModel):    x: timedelta
In [4]: %timeit Foo.parse_raw('{"x": "3e4"}')
14.7 µs ± 199 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

There's almost nothing between them so should be fine imo. :)

Reply:
Thank you for your feedback, I have added a try/catch in `isinstance(value, (str))` in the case of `value` is a `Dict`